packages/portal-marketmaker/src/index.js

```diff
@@ -0,0 +1,407 @@
+
+const provider = new ethers.providers.JsonRpcProvider(ethRpcProvider);
+const signer = new ethers.Wallet(privateKey, provider);
+
+const portal = new ethers.Contract(portalAddress, portalJSONABI, provider);
+const portalWithSigner = portal.connect(signer);
+
+const wbtc = new ethers.Contract(wbtcAddress, wbtcAbi, provider);
+const wbtcWithSigner = wbtc.connect(signer);
+
+
+var minblock = 0;
+
+function handlePendingOrder(order) {
+	var amountSats = BigInt(order.amountSats);
+	var priceTokPerSat = BigInt(order.priceTokPerSat);
+	if (amountSats > 0) {
+		bidSats += amountSats
+		bidLiquidity += amountSats * priceTokPerSat
+	} else {
+		askSats += amountSats
+		askLiquidity += amountSats * priceTokPerSat
+	}
+}
+
+async function loadPendingOrders() {
+	await axios.post('https://api.thegraph.com/subgraphs/name/kahuang/silver-portal', JSON.stringify({
+	    query : myOrdersQuery,
+	    variables : { maker, status },
+	  }), {
+	  headers: {
+	    'Content-Type': 'application/json',
+	    'Accept': 'application/json',
+	  }})
+	  .then(r => r.data.data.orders.map(order => handlePendingOrder(order)));
+}
+
+function sleep(ms) {
+	return new Promise((resolve) => {
+	  setTimeout(resolve, ms);
+	});
+}
+
+async function handlePendingEscrows(escrows) {
+	promises = [];
+	for (var i = 0; i < escrows.length; i ++ ) {
+		escrow = escrows[i];
+		console.log("Handline escrow: ", escrow);
+		var buf = Buffer.from(strip0x(escrow.destScriptHash), "hex");
+		var addr = bitcore.Address.fromScriptHash(buf, 'testnet', 'scripthash').toString();
+		const amountSatsDue = parseInt(escrow.amountSatsDue);
+		console.log("Sending ", amountSatsDue, " sats to ", addr);
+		const txid = (await sendBitcoin(addr, amountSatsDue)).txid;
```

Comment:
so... does `sendBitcoin` return a promise that waits for block confirmations? will resolve veeery slowly

Reply:
sendBitcoin sends the tx to the mempool and returns the transaction id. we wait for confirmations in the next part